ci/bazel-scripts/main.sh

```diff
@@ -11,16 +11,6 @@ ic_version_rc_only="0000000000000000000000000000000000000000"
 release_build="false"
 s3_upload="False"
 
-protected_branches=("master" "rc--*" "hotfix-*" "master-private")
-
-# if we are on a protected branch or targeting a rc branch we set ic_version to the commit_sha and upload to s3
-for pattern in "${protected_branches[@]}"; do
-    if [[ "$BRANCH_NAME" == $pattern ]]; then
-        IS_PROTECTED_BRANCH="true"
-        break
-    fi
-done
-
```

Comment:
Unfortunately and confusingly setting `IS_PROTECTED_BRANCH=${{ github.ref_protected }}` has an alternative to the above is not enough.

In ic-private the `hotifx-*` branch is not protected.  This means that if folks follow the "How to handle fixes for IC security vulnerabilities?" procedure, which specifies to push the hotfix to a `hotfix-*` branch and then expects the artifacts to be uploaded (`s3_upload="True"`), their artifacts won't be uploaded.

Reply:
@basvandijk arguably this is an issue in `ic-private` where `protected_branches` is then not aptly named, and/or an issue with the hotfix procedure. I'll see what I can do about that.

Reply:
Note that we shouldn't protect `hotfix-*` in ic-private since that would prevent folks from creating hotfixed.

So I think it's indeed a naming issue. Maybe we should use a different wordt then "PROTECTED" above.

Reply:
Maybe:

* `protected_branches` -> `release_branches`
* `IS_PROTECTED_BRANCH` -> `IS_RELEASE_BRANCH`